pkg/liqoctl/status/pods.go

```diff
@@ -0,0 +1,397 @@
+// Copyright 2019-2021 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"text/tabwriter"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	k8s "k8s.io/client-go/kubernetes"
+
+	"github.com/liqotech/liqo/pkg/utils/slice"
+)
+
+const (
+	checkerName = "liqo-control-plane"
+)
+
+var (
+	liqoDeployments = []string{
+		"capsule-controller-manager",
+		"liqo-controller-manager",
+		"liqo-network-manager",
+		"liqo-crd-replicator",
+		"liqo-webhook",
+		"liqo-gateway",
+		"liqo-auth",
+	}
+	liqoDaemonSets = []string{
+		"liqo-route",
+	}
+)
+
+// errorCount is a struct holding a list of errors.
+type errorCount struct {
+	errors []error
+}
+
+// collectionError struct holding the error for a given deployment of Liqo while collecting its status.
+type collectionError struct {
+	appType string
+	appName string
+	err     error
+}
+
+// newCollectionError return a new collectionError with the given arguments set.
+func newCollectionError(appType, appName string, err error) collectionError {
+	return collectionError{
+		appType: appType,
+		appName: appName,
+		err:     err,
+	}
+}
+
+// errorCountMap is a map with:
+// key -> name of the pod;
+// value -> errorCount for the given pod.
+type errorCountMap map[string]*errorCount
+
+// podState counts the number of pods in a particular state for a given deployment.
+type podState struct {
```

Comment:
```suggestion
type componentState struct {
```

Just a proposal though. I would personally avoid the term pod, since AFAIU here you are possibly referring to groups of pods (e.g., in case of daemonsets, or deployments with multiple replicas)